cloudbuild/snippets/quickstart_test.py

```diff
@@ -17,10 +17,14 @@
 
 import quickstart
 
+import time
+
 
 def test_quickstart(capsys: pytest.CaptureFixture) -> None:
     quickstart.quickstart()
     out, _ = capsys.readouterr()
+    # Waiting 5 sec
+    time.sleep(5)
```

Comment:
Py 3.11 tests failed with message:
```
AssertionError: assert 'Status.SUCCESS' in 'IN PROGRESS:
```

Is there a better way to wait for the build to finish instead of sleeping?

Reply:
I'd really prefer if we could implement some kind of waiting mechanism that queries the build status. Relying on `time.sleep()` in many places has already caused some mess in our tests.

Ideally, we want to have a `wait(timeout)` method to just wait for the build to be completed. If it fails in some time, we fail the test.

Reply:
I took a look at the [quickstart.py](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/cloudbuild/snippets/quickstart.py) file and I think it would be nice if it included a demo of polling the cloudbuild API for build status.

Reply:
@m-strzelczyk would using `backoff` as we suggest in our [authoring guide](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#retry-rpcs) fit with what you're thinking?

Reply:
The problem doesn't seem to be waiting long enough. See PR https://github.com/GoogleCloudPlatform/python-docs-samples/pull/9271/files. It waits for the operation to complete. But the string Status.SUCCESS isn't in the output, the value of that is ("3"). I think this can be closed unmerged and let #9271 deal with it.